src/components/tiles/table/use-data-set.ts

```diff
@@ -22,73 +22,91 @@ interface IUseDataSet {
+      const _selectedCell = dataSet.selectedCells[0];
+      selectedCellIndices.selectedCellColumnIndex = dataSet.attrIndexFromID(_selectedCell.attributeId) ?? -1;
+      selectedCellIndices.selectedCellRowIndex = _selectedCell.caseId === inputRowId.current
+        ? rows.length - 1
+        : dataSet.caseIndexFromID(_selectedCell.caseId);
+    }
+    return selectedCellIndices;
+  }
   const onSelectedCellChange = (position: TPosition) => {
-    const forward = (selectedCell.current.rowIdx < position.rowIdx) ||
-                    ((selectedCell.current.rowIdx === position.rowIdx) &&
-                      (selectedCell.current.idx < position.idx));
-    if ((position.rowIdx !== selectedCell.current.rowIdx) || (position.idx !== selectedCell.current.idx)) {
-      selectedCell.current = position;
+    // Only modify the selection if a single cell is selected
+    if (dataSet.selectedCells.length !== 1) return;
+
+    // Determine if we're moving forwards or backwards
+    const { selectedCellColumnIndex, selectedCellRowIndex } = getSelectedCellIndices();
+    const forward = (selectedCellRowIndex < position.rowIdx) ||
+      (selectedCellRowIndex === position.rowIdx && selectedCellColumnIndex < position.idx);
+
+    // Set the dataSet's selected cell
+    const newRowId = position.rowIdx === rows.length - 1
+      ? inputRowId.current
+      : dataSet.caseIDFromIndex(position.rowIdx);
+    const newColumnId = dataSet.attrIDFromIndex(position.idx - 1);
+    if (newColumnId && newRowId
+      && !(selectedCellColumnIndex === position.idx - 1 && selectedCellRowIndex === position.rowIdx)
+    ) {
+      dataSet.setSelectedCells([{ attributeId: newColumnId, caseId: newRowId }]);
       triggerRowChange();
     }
 
-    if (!isCellSelectable(position, columns, readOnly) && (columns.length > 2)) {
-      let newPosition = { ...position };
+    // Update the position if it's not a legal option (if we're in the control or delete column).
+    // Note that rdg will not allow us to move to a row outside of the grid
+    let newPosition = { ...position };
+    const rightColumnIndex = columns.length - (readOnly ? 1 : 2);
+    while(!isCellSelectable(newPosition, columns, readOnly)) {
       if (forward) {
-        while (!isCellSelectable(newPosition, columns, readOnly)) {
-          // move from last cell to { -1, -1 }
-          if ((newPosition.rowIdx >= rows.length) ||
-              ((newPosition.rowIdx === rows.length - 1) && (newPosition.idx >= columns.length - 1))) {
-            newPosition = { rowIdx: -1, idx: -1 };
-          }
-          // otherwise advance to next selectable cell
-          else if (++newPosition.idx >= columns.length) {
-            newPosition.idx = 1;
-            ++newPosition.rowIdx;
-          }
+        if (newPosition.rowIdx === rows.length - 1 && newPosition.idx > rightColumnIndex) {
+          // move from bottom right cell to top left cell
+          newPosition = { rowIdx: 0, idx: 1 };
```

Comment:
Is this a [keyboard trap](https://www.w3.org/TR/UNDERSTANDING-WCAG20/keyboard-operation-trapping.html)?

Reply:
Yes, yes it is. I thought this was how it was set up previously, but I now see it actually selected no cells in these cases. Thanks for pointing this out, I hadn't even considered how much of a problem this is!

Reply:
I've reverted the behavior back to what it was before these changes, which is just to deselect all selected cells when you reach before the start of the table or after the end of the table. However, I'm not sure if this actually fixes the accessibility issues here. I think the desired behavior would be to select the next or previous element (probably another tile, or maybe the table's toolbar?), but I think as is nothing is selected.

Reply:
It looks like RDG's [default](https://github.com/mui/mui-x/issues/4658) - and WCAG recommended - is that the tab key navigates _into_ and _out of_ the grid, and arrow keys are used for navigating _within_ the grid.

If that's not how it has worked in CLUE in the past, though, then there is a tradeoff between existing user expectations / convenience, and improving accessibility for keyboard users.

Since this is only one of many issues CLUE has with keyboard navigation, the practical impact of fixing this single issue is not large.